### PR TITLE
Connect React Redux with external API

### PR DESCRIPTION
**Pull request by emhamza**
**This is the most interesting milestone of the bookstore app so far.  I am happy that I have managed to complete the following tasks:**

- [x] Connect the Redux app with the bookstore API
- [x] Add functionality of get, post and delete book to my bookstore app

### CODE REVIEW

File: src/components/BookForm.js [hunk 1]
```diff
@@ -2,26 +2,37 @@ import React, { useState } from 'react';
...
         addBook({
           item_id: nanoid(),
           title,
           author,
           category: 'unknown', // typo fixed
         }),
       );
+      setError('');
       setTitle('');
       setAuthor('');
```
Comment (DammyShittu) [inline src/components/BookForm.js]: - [x] After adding a book it does not appear on the page immediately. I have to reload the browser for the newly added book to show up. Kindly note that when a book is added it should be displayed without having to reload the page. Kindly fix that.

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @emhamza,

## Highlights
✔️ API feature

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/BookForm.js (hunk 1 above)
Comment (emhamza) [reply src/components/BookForm.js]: Sure thing, thank  you so much for pointing that out! @DammyShittu

**Review by Davidosky007 — APPROVED** [review]
Hi @emhamza 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* API works fine 👍 
* Good looking readme 👍 
* Proper use of Gitflow 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**